Pod/Classes/XExtensionItem.h

```diff
@@ -96,10 +140,51 @@ typedef UIImage *(^XExtensionItemThumbnailProvidingBlock)(CGSize suggestedSize,
  */
 @property (nonatomic, copy) NSDictionary *userInfo;
 
+/**
+ Add entries from a dictionary-serializable custom object to this paramter object’s `userInfo` dictionary.
+ 
+ @param dictionarySerializable Object whose entries should be added to this paramter object’s `userInfo` dictionary.
+ */
+- (void)addEntriesToUserInfo:(id <XExtensionItemDictionarySerializing>)dictionarySerializable;
```

Comment:
That sounds like a really great solution, and `addCustomParameters:` seems like a good method name.